### PR TITLE
Refactor: Change variable names for schema and update docs

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Change variable names for schema and update docs

## Proposed changes

- Rename `avd_validate_conversion_mode` to `avd_data_conversion_mode`
- Rename `avd_validate_validation_mode` to `avd_data_validation_mode`
- Update schema docs with latest info
- Add clear warnings on disabling conversion.

### User Checklist

- N/A

### Repository Checklist

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md [hunk 1]
```diff
@@ -15,18 +15,22 @@ For details on the schema format see the [Schema Details](#schema-details) secti
 
 ## Automatic Type Conversion
 
-The validation of input variables is performed by the `arista.avd.validate` Action Plugin. The plugin also supports automatic
-conversion of variable types, if configured in the schema. The supported types that can be converted from depends on the type to
-be converted to. The current implementation supports the following automatic conversions:
+The validation of input variables is performed in most AVD Action Plugins. The validation also supports automatic conversion of
+variable types, if configured in the schema. The supported types that can be converted from depends on the type to be converted to.
+The current implementation supports the automatic conversions listed below. Conversions will only happen if set in the schema.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/input-variable-validation-BETA.md]: This `Conversions will only happen if set in the schema` is redundant with this `The validation also supports automatic conversion of
variable types, if configured in the schema.`

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md [hunk 2]
```diff
@@ -15,18 +15,22 @@ For details on the schema format see the [Schema Details](#schema-details) secti
...
 | From (`convert_types`) | To (`type`) |
 | ---------------------- | ----------- |
 | `bool`, `str` | `int` |
 | `int`, `str` | `bool` |
 | `bool`, `int` | `str` |
-| `dict`* | `list` |
+| `dict`*, `list`** | `list` |
 
-\* Conversion from `dict` to `list` is only supported when `primary_key` is set on the `list` schema.
+\* If `primary_key` is set on the `list` schema, conversion from `dict`-of-`dicts` to `list`-of-`dicts` will insert the `primary_key`
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/input-variable-validation-BETA.md:29]: Probably worth adding examples here

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md [hunk 3]
```diff
@@ -56,34 +60,36 @@ tenants:
 
 The input data will be converted automatically and validated according to the new model to allow for a gracious introduction of this data-model changes in late 3.x versions.
 
-Type conversion is turned on by default but can be disabled or adjusted with the options described in the next section.
+Type conversion is turned on by default.
+
+!!! CAUTION
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/input-variable-validation-BETA.md:65]: I am wondering if this is wise to even allow it given the drawbacks :) ..

**Review by gmuloc — APPROVED** [review]
LGTM - only cosmetic comments / remarks

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/input-variable-validation-BETA.md:65]: We have a few special use cases where disabling could make sense, but we could keep it hidden?

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/input-variable-validation-BETA.md:29]: That is the example below this section.

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/input-variable-validation-BETA.md]: Yeah I will clean it up

File: ansible_collections/arista/avd/docs/input-variable-validation-BETA.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/input-variable-validation-BETA.md:65]: we can revisit if we see problems

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**